### DISCUSSION TITLE
restart server on some event

### DISCUSSION THREAD

**Discussion by AlexKovalevych** [actix/actix-web#1940] [Q&A] [answered]
Is there any way to restart a web server (with graceful shutdown) after some event happened, similar to how it's done in tonic? https://docs.rs/tonic/0.4.0/tonic/transport/server/struct.Router.html#method.serve_with_shutdown

**Comment by fakeshadow** [comment, marked as answer]
https://docs.rs/actix-server/1.0.4/actix_server/struct.Server.html

**Reply by AlexKovalevych** [reply]
thank you
